### PR TITLE
Use `wp_cache_flush_runtime` in unit tests. 

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket:  https://core.trac.wordpress.org/ticket/31463

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/includes/abstract-testcase.php [hunk 1]
```diff
@@ -387,12 +387,9 @@ protected function _restore_hooks() {
 	public static function flush_cache() {
 		global $wp_object_cache;
 
-		$wp_object_cache->group_ops      = array();
-		$wp_object_cache->stats          = array();
-		$wp_object_cache->memcache_debug = array();
```
Comment (spacedmonkey) [inline tests/phpunit/includes/abstract-testcase.php:392]: These three properties are not used for anything. I have removed them for now.

File: tests/phpunit/includes/abstract-testcase.php [hunk 2]
```diff
@@ -387,12 +387,9 @@ protected function _restore_hooks() {
...
-		$wp_object_cache->cache          = array();
+		wp_cache_flush_runtime();
 
-		if ( method_exists( $wp_object_cache, '__remoteset' ) ) {
+		if ( is_object( $wp_object_cache ) && method_exists( $wp_object_cache, '__remoteset' ) ) {
```
Comment (spacedmonkey) [inline tests/phpunit/includes/abstract-testcase.php:392]: Not all object caches use this global. Check it is object before running this.

**Review by johnbillion — APPROVED** [review]

**Comment by tillkruss** [comment]
LGTM

**Review by tillkruss — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Committed https://core.trac.wordpress.org/changeset/55741

**Pull request closed without merging**